utils/build-script-impl

```diff
@@ -1252,6 +1252,7 @@ LIBDISPATCH_SOURCE_DIR="${WORKSPACE}/swift-corelibs-libdispatch"
 LIBDISPATCH_STATIC_SOURCE_DIR="${WORKSPACE}/swift-corelibs-libdispatch"
 LIBICU_SOURCE_DIR="${WORKSPACE}/icu"
 LIBCXX_SOURCE_DIR="${WORKSPACE}/llvm-project/libcxx"
+EXPERIMENTAL_STRING_PROCESSING_SOURCE_DIR="${WORKSPACE}/swift-experimental-string-processing"
```

Comment:
Why do these sources need to be in a separate repo?

Reply:
We plan to keep string processing in a package outside of the Swift repo for quicker development iterations and better modularity.  The _MatchingEngine target is both as a libswift library (used by the parser) and as a runtime library.  When it gets closer to the release we can consider moving it into the Swift repo.

Do you have concerns about this approach?

Reply:
You should really keep that sources inside the swift repo.

With "quicker development iteration", I guess you mean you want to avoid the overhead of the swift PR testing. But you'll have lit tests for that feature in the swift repo. This means, it will increase the probability that you'll break the swift lit tests. And that will add more troubles for everyone else.
Also, making swift depend on additional repos adds build complexity for all developers. Currently swift only strictly depends on llvm-project - and that's great. For example, most of the time I just update the swift repo to reduce my local build time.

Reply:
We would integrate based on a tag, so we'd do integration testing the PR that updates the tag. Beyond that, it doesn't make senes to do full integration testing as part of a normal development cycle.

Reply:
> But you'll have lit tests for that feature in the swift repo.

We have unit tests, functional tests, and integration tests. We use lit for integration testing.

Reply:
Still, I predict that this creates more troubles than it solves.
E.g. whenever you need to fix a bug, you have to update two repos: in `swift-experimental-string-processing` and in `swift` to update the tag. 
Also, it just complicates the build process, which increases the probability of running into problems - as you can already see yourself: the Windows PR testing.